projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -394,9 +397,93 @@ internal void OnCallbackException(CallbackExceptionEventArgs args)
             _callbackExceptionWrapper.Invoke(this, args);
         }
 
-        internal void Write(ReadOnlyMemory<byte> memory)
+        internal void Write(ReadOnlySpan<byte> payload)
+        {
+            if (!_frameHandler.IsClosed)
+            {
+                _writeSemaphore.Wait();
+                WriteSpan(payload);
+                _frameHandler.FrameWriter.FlushAsync().AsTask().Wait(); // Sync-over-async, not great, not terrible. All we can do for the sync path.
```

Comment:
I have not yet used the pipeline stuff very intensively. I was wondering if we could use `AsStream` and then use the synchronous stream API here and in the other sync parts?

Reply:
That actually makes some sense. I'll look into that.